packages/server/graphql/public/mutations/startRecurrence.ts

```diff
@@ -37,37 +40,88 @@ const startRecurrence: MutationResolvers['startRecurrence'] = async (
     return standardError(new Error('Meeting is not a team prompt meeting'), {userId: viewerId})
   }
 
-  if (meeting.meetingSeriesId) {
-    return standardError(new Error('Meeting already has meeting series'), {userId: viewerId})
-  }
-
   // Next meeting start is tomorrow at 9a UTC
   // :TODO: (jmtaber129): Determine this from meeting series configuration.
-  const startDate = new Date(
+  const nextMeetingStartDate = new Date(
     Date.UTC(now.getUTCFullYear(), now.getUTCMonth(), now.getUTCDate() + 1, 9)
   )
   const recurrenceRule = new RRule({
     freq: RRule.WEEKLY,
     byweekday: [RRule.MO, RRule.TU, RRule.WE, RRule.TH, RRule.FR],
-    dtstart: startDate
+    dtstart: nextMeetingStartDate
   })
 
-  const newMeetingSeriesId = await insertMeetingSeriesQuery({
+  if (meeting.meetingSeriesId) {
+    const meetingSeries = await dataLoader.get('meetingSeries').loadNonNull(meeting.meetingSeriesId)
+    const {cancelledAt, recurrenceRule, id: meetingSeriesId} = meetingSeries
+    if (!cancelledAt) {
+      return standardError(new Error('Meeting is already recurring'), {userId: viewerId})
+    }
+
+    const currentRRule = RRule.fromString(recurrenceRule)
+    const hasRecentlyStopped = cancelledAt.getTime() > now.getTime() - ms('1h')
+    const nextScheduledStartDate = currentRRule.after(now)
+    const isNextScheduledStartDateReasonable =
+      nextScheduledStartDate.getTime() > now.getTime() + ms('9h')
+    if (hasRecentlyStopped || isNextScheduledStartDateReasonable) {
+      // no need to change the recurrence rule, next meeting starts in reasonable time
+      await restartMeetingSeries({cancelledAt: null}, meetingSeriesId)
+    } else {
+      // meeting series was stopped more than ~1h ago and the next scheduled start date is within the next ~9h
+      // to prevent a meeting from being scheduled too soon, we need to change the recurrence rule
+      const newRecurrenceRule = currentRRule.clone()
+      newRecurrenceRule.options.dtstart = new Date(nextMeetingStartDate.getTime() + ms('1d'))
+
+      await restartMeetingSeries(
+        {cancelledAt: null, recurrenceRule: newRecurrenceRule.toString()},
+        meetingSeriesId
+      )
+    }
+
+    dataLoader.get('meetingSeries').clear(meetingSeriesId)
+
+    // update all the active meetings to end at proper time
+    const activeMeetings = await r
+      .table('NewMeeting')
+      .getAll(meetingSeriesId, {index: 'meetingSeriesId'})
+      .filter({endedAt: null}, {default: true})
+      .run()
+    const updates = activeMeetings.map((meeting) =>
+      r
+        .table('NewMeeting')
+        .get(meeting.id)
+        .update({
+          scheduledEndTime: new Date(
+            meeting.createdAt.getTime() + ms(`${MEETING_DURATION_IN_MINUTES}m`)
```

Comment:
+1 This should use the `meetingSeries`' `duration`

Reply:
good catch!